src/include/common/json.h

```diff
@@ -4,22 +4,23 @@
 #include "nlohmann/json.hpp"
 
 namespace terrier::common {
+// TODO(WAN): I think we should be using the adt_serializable, not this hack..
```

Comment:
Yeah i agree, we should offload as much the json responsibility to some library that is meant to maintain it. Can you open an issue for this. Would help us avoid the difficulties you're facing right now in the future.